### ISSUE TITLE
PersistCookieJar gives an error after .deleteAll is used

### ISSUE THREAD

**Issue by MichaelPriebe** [flutterchina/dio#171]
## Steps to Reproduce

```dart
PersistCookieJar cookieJar = PersistCookieJar();
cookieJar.deleteAll();
cookieJar.saveFromResponse(
  Uri.parse('https://flutter.io/'),
  [
    Cookie('test', 'cookie'),
  ],
);
```
## Logs

FileSystemException: Cannot open file

**Comment by wendux** [comment]
In flutter, the path passed to PersistCookieJar must be valid(exists in phones and with write access). you can use path_provider package to get right path.

**Comment by MichaelPriebe** [comment]
Yes I already use path_provider. It seems Dio deletes the file with delete all and when it goes to save it, it doesn't exist!

**Comment by wendux** [comment]
fixed in 2.0.7 version, please update and  try again.
